CenterForOpenScience/ember-osf-web
# [ENG-1611] Update instutional dashboard

- Ticket: [ENG-1611]
- Feature flag: n/a

## Purpose
Update institutional dashboard to match what is implemented in the API

## Summary of Changes
- Added models `institution-department` and `institution-summary-metric`
- Updated model `institutional-user` to `institution-user`
- Update model for institution route to return institution, users, departments, and summary metrics
- Update mirage to handle these new relationships on `institution`
- Update `institution/<id>/dashboard` and components within it to use these new models

## Side Effects
`osf-model` and `capture-exception` has been updated. An error was getting thrown when trying to get the related counts for `user-metrics` on the institution, stopping the rendering of the page.  Since related counts are not going to be added to `institutionUsers` relationship on Institution Detail endpoint (Platform is "avoiding mixing ElasticSearch and Database calls in same view"), the error had to be captured, instead of thrown. Not sure if this will cause issues/side-effects.

## QA Notes
- Please ensure the `/institutions/:id/dashboard` route is served ([ENG-1823])
 - only admins should be able to access this route, and non-admins should be rerouted to our `not-found` page
- Users should be sortable by name and department
- Users should be filterable by department
- Institution info should be exportable as a CSV file ([ENG-1810])
- Total number of Single Sign-On users should be displayed
- Total number of public and private projects in the institution should be shown
- Pie chart of users divided by department should be shown


[ENG-1611]: https://openscience.atlassian.net/browse/ENG-1611

[ENG-1823]: https://openscience.atlassian.net/browse/ENG-1823
[ENG-1810]: https://openscience.atlassian.net/browse/ENG-1810
